pkg/binary/binary.go

```diff
@@ -18,21 +21,22 @@ type Callback func(*Binary) (string, error)
 type Binary struct {
 	Context context.Context `json:"-"`
 	// for installation
-	URL           string            `json:"-"`
-	URLF          Callback          `json:"-"`
-	GitHubRepo    string            `json:"repo"`
-	GitHubFile    string            `json:"-"`
-	GitHubFileF   Callback          `json:"-"`
-	Version       string            `json:"-"`
-	VersionF      Callback          `json:"-"`
-	VersionLocalF Callback          `json:"-"`
-	Name          string            `json:"name" yaml:"name"`
-	File          string            `json:"-"`
-	IsTarGz       bool              `json:"-"`
-	IsZip         bool              `json:"-"`
-	TarFile       string            `json:"-"`
-	TarFileF      Callback          `json:"-"`
-	Tracker       *progress.Tracker `json:"-"`
+	URL           string          `json:"-"`
+	URLF          Callback        `json:"-"`
+	GitHubRepo    string          `json:"repo"`
+	GitHubFile    string          `json:"-"`
+	GitHubFileF   Callback        `json:"-"`
+	Version       string          `json:"-"`
+	VersionF      Callback        `json:"-"`
+	VersionLocalF Callback        `json:"-"`
+	Name          string          `json:"name" yaml:"name"`
+	File          string          `json:"-"`
+	IsTarGz       bool            `json:"-"`
+	IsZip         bool            `json:"-"`
+	TarFile       string          `json:"-"`
+	TarFileF      Callback        `json:"-"`
+	Tracker       *pretty.Tracker `json:"-"`
+	Writer        *pwrap.Writer   `json:"-"`
```

Comment:
You don't need `Writer`
`Tracker` has the `Writer` instance.

https://github.com/fentas/goodies/blob/main/progress/progress.go#L32C2-L34C3

You can use `Tracker` as it inherits everything from `Writer`.

https://github.com/fentas/goodies/blob/main/progress/progress.go#L10-L12

So `b.Tracker` refers to `b.Writer` or if you want to be more explicit you can also write `b.Tracker.Writer`
But in the end, `b.Tracker` points to `b.Tracker.Writer`.